xbmc/windowing/android/AndroidUtils.cpp

```diff
@@ -348,3 +372,38 @@ void CAndroidUtils::OnSettingChanged(const std::shared_ptr<const CSetting>& sett
   if (settingId == CAndroidUtils::SETTING_LIMITGUI)
     CDisplaySettings::GetInstance().ClearCalibrations();
 }
+
+const std::vector<int> CAndroidUtils::GetDisplaySupportedHdrTypes()
+{
+  CJNIWindow window = CXBMCApp::getWindow();
+
+  if (window)
+  {
+    CJNIView view = window.getDecorView();
+    if (view)
+    {
+      CJNIDisplay display = view.getDisplay();
+      if (display)
+      {
+        CJNIDisplayHdrCapabilities caps = display.getHdrCapabilities();
+        if (caps)
+          return caps.getSupportedHdrTypes();
+      }
+    }
+  }
+
+  return {};
+}
+
+const void CAndroidUtils::LogDisplaySupportedHdrTypes()
+{
+  const std::vector<int> hdrTypes = GetDisplaySupportedHdrTypes();
+  std::stringstream ss;
+
+  for (int type : hdrTypes)
+  {
+    ss << HdrTypeString(type) << " ";
+  }
+
+  CLog::Log(LOGDEBUG, "CAndroidUtils: Display supported HDR types: {}", ss.str());
```

Comment:
here you are working with strings, you can avoid to use stringstream and use a simple std::string

Reply:
That's true. Initially I was trying to make it prettier with a separator.